<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Exposomic studies | Omic Association Studies with R and Bioconductor</title>
  <meta name="description" content="9 Exposomic studies | Omic Association Studies with R and Bioconductor" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Exposomic studies | Omic Association Studies with R and Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Exposomic studies | Omic Association Studies with R and Bioconductor" />
  
  
  

<meta name="author" content="Juan R González &amp; Alejandro Cáceres" />


<meta name="date" content="2020-05-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="epigenomic-studies.html"/>
<link rel="next" href="enrichment-analysis.html"/>
<script src="libs/header-attrs-2.1/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />












</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#book-summary"><i class="fa fa-check"></i>Book summary</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#book-overview"><i class="fa fa-check"></i>Book overview</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#datasets-used-in-the-book"><i class="fa fa-check"></i>Datasets used in the book</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the authors</a></li>
<li class="chapter" data-level="" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="case-examples.html"><a href="case-examples.html"><i class="fa fa-check"></i><b>2</b> Case examples</a></li>
<li class="chapter" data-level="3" data-path="dealing-with-omic-data-in-bioconductor.html"><a href="dealing-with-omic-data-in-bioconductor.html"><i class="fa fa-check"></i><b>3</b> Dealing with <em>omic</em> data in Bioconductor</a></li>
<li class="chapter" data-level="4" data-path="genetic-association-studies.html"><a href="genetic-association-studies.html"><i class="fa fa-check"></i><b>4</b> Genetic association studies</a></li>
<li class="chapter" data-level="5" data-path="genomic-variant-studies.html"><a href="genomic-variant-studies.html"><i class="fa fa-check"></i><b>5</b> Genomic variant studies</a></li>
<li class="chapter" data-level="6" data-path="addressing-batch-effects.html"><a href="addressing-batch-effects.html"><i class="fa fa-check"></i><b>6</b> Addressing batch effects</a></li>
<li class="chapter" data-level="7" data-path="transcriptomic-studies.html"><a href="transcriptomic-studies.html"><i class="fa fa-check"></i><b>7</b> Transcriptomic studies</a></li>
<li class="chapter" data-level="8" data-path="epigenomic-studies.html"><a href="epigenomic-studies.html"><i class="fa fa-check"></i><b>8</b> Epigenomic studies</a></li>
<li class="chapter" data-level="9" data-path="exposomic-studies.html"><a href="exposomic-studies.html"><i class="fa fa-check"></i><b>9</b> Exposomic studies</a></li>
<li class="chapter" data-level="10" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html"><i class="fa fa-check"></i><b>10</b> Enrichment analysis</a></li>
<li class="chapter" data-level="11" data-path="multiomic-data-analysis.html"><a href="multiomic-data-analysis.html"><i class="fa fa-check"></i><b>11</b> Multiomic data analysis</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Omic Association Studies with R and Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exposomic-studies" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Exposomic studies</h1>
<p>In this chapter, we study exposomic data. This type of data is different from the <em>omic</em> data that we discussed in the previous chapters. To start, the exposure variables that constitute the dataset are measured with different experimental techniques. They are highly diverse and heterogeneous, as they may include biological assays, obtained from personal questionnaires or derived from inferred exposures at geolocalized sites. Nevertheless, organisms respond to environmental changes and their efficiency in the response may result in phenotypic differences that may lead to adaptation or disease. The identification of the environmental changes is therefore essential to identify phenotypic differences among individuals and an unbiased scan on a large number of exposures is a starting point. In addition, exposure data may affect other molecular processes such as gene transcription or methylation. And therefore, important mechanisms may be discovered from the association between exposomic and transcriptomic, or methylomic data. Integration of <em>omic</em> data is then required. From the analysis perspective, exposomic data can be treated as the previous <em>omic</em> data, in which data structures are defined and methods are implemented on them. Here, we illustrate the use of <code>rexposome</code>, a comprehensive Bioconductor package with functions to impute, normalize and characterize correlational structures in exposomic data, to perform exposome-wise association analysis (ExWAS) and to test the association between exposomic and other <em>omic</em> variables.</p>
<p>The <code>R</code> code to reproduce the results and figures of this chapter can be found <a href="https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/ch8.R">here</a>.</p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="epigenomic-studies.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="enrichment-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
