---
title: "Omic Association Studies with R and Bioconductor"
author: "Juan R Gonzalez & Alejandro Caceres"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
output:
  bookdown::gitbook: default
  #bookdown::pdf_book: default
---


# Preface {-}

![](figures/cover_book.png)


The online material of this book is licensed under the Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License. You can purchase a hardcopy from [CRC Press](https://www.crcpress.com/Omic-Association-Studies-with-R-and-Bioconductor/Gonzalez-Caceres/p/book/9781138340565) or [Amazon](https://www.amazon.es/Omic-Association-Studies-R-Bioconductor/dp/1138340561).


## Book overview {-}


This book is concerned with the analysis of high dimensional data that is acquired at specific biological domains. The aim of the analyses is the explanation of phenotypic differences among individuals. We, therefore, search for endogenous and exogenous factors that may influence such differences. The endogenous domains on which we turn our attention are those at the molecular level involving basic DNA structure and function, which have been labeled with the _omic_ suffix. In particular, we will describe current methods to analyze genomic data, which is high-dimensional at the gene (DNA sequence) level, transcriptomic data involving transcription of DNA into mRNA and epigenomic/methylomic data that relate to the epigenetic modifications of DNA. Many of the methods used at each domain overlap due to the biological nature and high dimensionality of data. However, important specificities remain, some derived from the acquisition of data and others from differences in the underlying biological processes. Within the exogenous domain, we study the high dimensional acquisition of exposure factors that are believed to influence the development or progression of individual traits.    

## Install required packages {-}

## Datasets used in the book {-}




# Introduction

This chapter provides an overview of _omic_ data as well as the different types of association studies that can be carried out to study the impact of those _omic_ datasets on complex diseases, quantitative traits or any other coditions. Tne chapter continues by providing an outline of the different publicy available databases that allow the users to analyze different _omic_ data. It is illustrated how these data can be accessed using `R`. Finally, Bioconductor an `R` are briefly introduced. 

The `R` code to reproduce the results and figures of this chapter can be found [here](https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/ch1.R).

# Case examples

In this chapter, we will show, with five case examples, how to retrieve data from five public repositories and some basic analysis that can be performed on the data.  We introduce the functions and packages in `R`/Bioconductor that can be used to perform specific queries, retrieval, and analysis, all within a single `R` session. Further chapters will treat in detail the packages and the functions used to produce the results. 

The `R` code to reproduce the results and figures of this chapter can be found [here](https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/ch2.R).


# Dealing with _omic_ data in Bioconductor

This chapter offers a summary of the main data structures that are implemented in Bioconductor for dealing with genomic, transcriptomic, methylomic and exposomic data. The structures are objects to which methods are applied. _Omic_ data is typically composed of three datasets: one containing the actual high-dimensional data of _omic_ variables per individuals, annotation data that specifies the characteristics of the variables and phenotypic information that encodes the subject's traits of interest, covariates and sampling characteristics. For instance, transcriptomic data is stored in a `ExpressionSet` object, which is a data structure that contains the transcription values of individuals at each transcription probe, the genomic information for the transcription probes and the phenotypes of the individuals. Specific data is accessed, processed and analyzed with specific functions from diverse packages, conceived as methods acting on the `ExpressionSet` object. The aim of this chapter is then to introduce the specific _omic_ objects available in Bioconductor. In the following chapters, we will introduce the packages that have been implemented to process and analyze these objects.     
 

The `R` code to reproduce the results and figures of this chapter can be found [here](https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/ch3.R).




# Genetic association studies

Genetic association studies aim to identify genetic variants that explain subject differences in qualitative or quantitative traits. Genetic variants may include SNPs, CNVs, mosaicisms or polymorphic inversions, which can be obtained from SNP array data. In this chapter, we thoroughly discuss SNP association studies. We explain the different genetic models to be used in single SNP associations and how they are applied in a massive univariate testing when several SNPs are analyzed. Relating the simultaneous analysis of multiple SNPs, we discuss haplotype and genetic scores associations. We then move to the analysis of genome-wide association studies, where high-dimensional genomic data is treated. Issues concerning quality control of SNPs and individuals are discussed, as well as association tests and possible sources of confounding such as population stratification. We finish SNP analyses with tools that help to interpret results. 

The `R` code to reproduce the results and figures of this chapter can be found [here](https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/ch4_SNPs.R).




# Genomic variant studies

The `R` code to reproduce the results and figures of this chapter can be found [here](https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/ch4_CNVs.R).



# Addressing batch effects

The `R` code to reproduce the results and figures of this chapter can be found [here](https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/ch6.R).



# Transcriptomic studies

The `R` code to reproduce the results and figures of this chapter can be found [here](https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/ch7.R).



# Epigenomic studies

The `R` code to reproduce the results and figures of this chapter can be found [here](https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/ch8.R).



# Exposomic studies

The `R` code to reproduce the results and figures of this chapter can be found [here](https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/ch9.R).



# Enrichment analysis

The `R` code to reproduce the results and figures of this chapter can be found [here](https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/ch10.R).


# Multiomic data analysis

The `R` code to reproduce the results and figures of this chapter can be found [here](https://github.com/isglobal-brge/book_omic_association/blob/master/R_chunks/c11.R).