<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Omic Association Studies with R and Bioconductor</title>
  <meta name="description" content="Omic Association Studies with R and Bioconductor" />
  <meta name="generator" content="bookdown 0.11 and GitBook 2.6.7" />

  <meta property="og:title" content="Omic Association Studies with R and Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Omic Association Studies with R and Bioconductor" />
  
  
  

<meta name="author" content="Juan R González &amp; Alejandro Cáceres" />


<meta name="date" content="2019-06-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="about-the-authors.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#book-summary"><i class="fa fa-check"></i>Book summary</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#book-overview"><i class="fa fa-check"></i>Book overview</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#install-required-packages"><i class="fa fa-check"></i>Install required packages</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#datasets-used-in-the-book"><i class="fa fa-check"></i>Datasets used in the book</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the authors</a></li>
<li class="chapter" data-level="" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="case-examples.html"><a href="case-examples.html"><i class="fa fa-check"></i><b>2</b> Case examples</a></li>
<li class="chapter" data-level="3" data-path="dealing-with-omic-data-in-bioconductor.html"><a href="dealing-with-omic-data-in-bioconductor.html"><i class="fa fa-check"></i><b>3</b> Dealing with <em>omic</em> data in Bioconductor</a></li>
<li class="chapter" data-level="4" data-path="genetic-association-studies.html"><a href="genetic-association-studies.html"><i class="fa fa-check"></i><b>4</b> Genetic association studies</a></li>
<li class="chapter" data-level="5" data-path="genomic-variant-studies.html"><a href="genomic-variant-studies.html"><i class="fa fa-check"></i><b>5</b> Genomic variant studies</a></li>
<li class="chapter" data-level="6" data-path="addressing-batch-effects.html"><a href="addressing-batch-effects.html"><i class="fa fa-check"></i><b>6</b> Addressing batch effects</a></li>
<li class="chapter" data-level="7" data-path="transcriptomic-studies.html"><a href="transcriptomic-studies.html"><i class="fa fa-check"></i><b>7</b> Transcriptomic studies</a></li>
<li class="chapter" data-level="8" data-path="epigenomic-studies.html"><a href="epigenomic-studies.html"><i class="fa fa-check"></i><b>8</b> Epigenomic studies</a></li>
<li class="chapter" data-level="9" data-path="exposomic-studies.html"><a href="exposomic-studies.html"><i class="fa fa-check"></i><b>9</b> Exposomic studies</a></li>
<li class="chapter" data-level="10" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html"><i class="fa fa-check"></i><b>10</b> Enrichment analysis</a></li>
<li class="chapter" data-level="11" data-path="multiomic-data-analysis.html"><a href="multiomic-data-analysis.html"><i class="fa fa-check"></i><b>11</b> Multiomic data analysis</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Omic Association Studies with R and Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Omic Association Studies with R and Bioconductor</h1>
<p class="author"><em>Juan R González &amp; Alejandro Cáceres</em></p>
<p class="date"><em>2019-06-16</em></p>
</div>
<div id="preface" class="section level1 unnumbered">
<h1>Preface</h1>
<p><img src="figures/cover_book.png" /></p>
<p>The online material of this book is licensed under the Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License. You can purchase a hardcopy from <a href="https://www.crcpress.com/Omic-Association-Studies-with-R-and-Bioconductor/González-Cáceres/p/book/9781138340565">CRC Press</a> or <a href="https://www.amazon.es/Omic-Association-Studies-R-Bioconductor/dp/1138340561">Amazon</a>.</p>
<div id="book-summary" class="section level2 unnumbered">
<h2>Book summary</h2>
<p>After the great expansion of genome-wide association studies, their scientific methodology and, notably, their data analysis has matured in recent years, and they are a keystone in large epidemiological studies. Newcomers to the field are confronted with a wealth of data, resources and
methods. This book presents current methods to perform informative analyses using real and illustrative data with established bioinformatics tools and guides the reader through the use of publicly available data. Includes clear, readable programming codes for readers to reproduce and
adapt to their own data. Emphasises extracting biologically meaningful associations between traits of interest and genomic, transcriptomic and epigenomic data</p>
<ul>
<li>Uses up-to-date methods to exploit omic data</li>
<li>Presents methods through specific examples and computing sessions</li>
<li>Supplemented by a website, including code, datasets, and solutions</li>
</ul>
</div>
<div id="book-overview" class="section level2 unnumbered">
<h2>Book overview</h2>
<p>The aim of the book is to offer a practical guide to researchers, graduate students and those interested in the analysis of <em>omic</em> data. While our emphasis is on the use of data in publicly available repositories, the reader interested in analyzing novel data will find settled methods for inquiring into high-dimensional biological data. We have conceived the book as a first reference to tackle specific types of data, as well as a textbook for a bioinformatics course at the MSc level. Our objective is to demonstrate how to analyze genomic, transcriptomic, epigenomic and exposomic data to explain phenotypic differences among individuals. We describe the first analyses and methods of inquiry that should be used to identify the patterns in the data that associate with a trait of interest. During the past decade numerous methods have been developed and, due to the complexity of the data, we expect many more to be devised. Nonetheless, we describe some of the most established methods that are available in the Bioconductor and <code>R</code> repositories, which should constitute the first line of inquiry and to which future developments should be compared against.</p>
<p>The methods and applications described here are all publicly available and are accessible to anyone comfortable with fitting a linear regression model in <code>R</code>. While we direct the reader to numerous introductory books in <code>R</code> and basic statistical methods, the present book is directed to users. From a basic user level, we aim to guide the readers to expand their toolkit in order to deal with <em>omic</em> data with confidence.</p>
<p>All the methods discussed here are part of our daily toolkit. We are regular users of all the methods and are also developers of many of them. The book is the result of compiling workshop and class material, of software package development and of years of research carried out in Juan R. González’s Bioinformatics Group in Genetic Epidemiology, within ISGlobal. We have thus developed expertise in the use of the methods and in their communication, and have realized the need to offer a guide to new researchers in the field. There is a wealth of publicly available software and data, yet the landscape is overwhelming to newcomers. We offer them starting points from which to begin inquiring into the <em>omic</em> data of interest. We do not offer a complete or global view but indicate safe up-to-date entry points. As developers of some of the packages discussed, we are committed, as part of the Bioconductor community, to offer clear and reproducible documentation, clarify doubts and update new versions. We insist that packages and pipelines to assist users are also implemented so they are further improved by other developers.</p>
<p>The material discussed in the book is largely based on cheap high-throughput methods. They include microarrays and some sequencing methods such as RNA-sequencing. We are also aware of the developments in the collection of new high-dimensional biological data, such as Next-Generation Sequencing or those aimed at single cells. There are, however, important advantages in the use and analysis of microarrays which will keep them relevant for many years. First, association studies require cohorts and technologies to be scalable to hundreds of thousands of individuals to properly power epidemiological inferences. Microarrays clearly meet the target. While we may conceive such scalability for future sequencing, the preprocessing of data may change but the basic methods of inference would likely remain the same. In addition, microarray data is widely available and it has been an important source of continuous reanalysis to test novel focused hypotheses, confirm new results or reproduce previous findings. Finally, SNPs arrays can be additionally used to explore other genomic variants, for which specific high-throughput technology is not yet available. Therefore, association analyses in large cohorts can be performed on inversion polymorphisms and mosaicism, including the loss of chromosome Y.</p>
</div>
<div id="install-required-packages" class="section level2 unnumbered">
<h2>Install required packages</h2>
</div>
<div id="datasets-used-in-the-book" class="section level2 unnumbered">
<h2>Datasets used in the book</h2>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="about-the-authors.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
